AG-11385 Use aria-disabled="true" instead of disabled

olegat opened the pull request:
https://ag-grid.atlassian.net/browse/AG-11385

The guideline is to make buttons of toolbars/menus focusable even when they are disabled. This is so that the buttons are discoverable to screenreader users.

Reference:
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/menu_role
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/roles/toolbar_role

olegat committed:
AG-11385 Use aria-disabled="true" instead of disabled

The guideline is to make buttons of toolbars/menus focusable even when
they are disabled. This is so that the buttons are discoverable to
screenreader users.

Reference:
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/menu_role
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/roles/toolbar_role

lsjroberts reviewed (commented):
If I understand this correctly - https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-disabled

I think we'll need to add a check on the `onclick` for the button's `ariaDisabled === 'true'` to stop it from firing?

olegat committed:
AG-11385 Add noop onclick listeners to aria-disabled buttons

olegat committed:
AG-11385 Ignore clicks on focus-disabled buttons in InteractionManager

olegat committed:
AG-11385 Only fire 'drag-start' when the main button is pressed

This fixes a bug where the a 'grabbing' cursor is used when showing the
contextmenu with Zoom enabled.

olegat committed:
nx format

olegat commented:
I've added `onclick` listeners to `aria-disabled="true"` buttons.

lsjroberts commented on packages/ag-charts-enterprise/src/features/context-menu/contextMenu.ts:
```diff
@@ -276,10 +277,14 @@ export class ContextMenu extends _ModuleSupport.BaseModuleInstance implements _M
-            return this.createDisabledElement(label);
+            button = this.createDisabledElement(label);
+        } else {
+            button = this.createButtonElement(label);
```
nit: could merge `createButtonElement` and `createDisabledElement` as they're so similar? Then add the `ariaDisabled` in here if disabled?

olegat replied:
Done.

lsjroberts reviewed (approved):
LGTM

olegat committed:
AG-11385 CR: Merge createButtonElement and createDisabledElement

olegat committed:
AG-11385 Update snapshot
